coderd/database/migrations/000258_add_otp_reset_to_user.up.sql

```diff
@@ -0,0 +1,8 @@
+ALTER TABLE users ADD COLUMN hashed_one_time_passcode bytea;
+COMMENT ON COLUMN users.hashed_one_time_passcode IS 'A hash of the one-time-passcode given to the user.';
+
+ALTER TABLE users ADD COLUMN one_time_passcode_expires_at timestamp with time zone;
+COMMENT ON COLUMN users.one_time_passcode_expires_at IS 'The time when the one-time-passcode expires.';
+
+ALTER TABLE users ADD COLUMN must_reset_password bool NOT NULL DEFAULT false;
```

Comment:
I also think you could add a `CONSTRAINT` here such that when this field is set to `true` the OTP columns must be set.

Reply:
I probably should have written a better description of how I see this working.

1. User enters email to request a one-time-passcode.
  - `one_time_passcode_expires_at` and `hashed_one_time_passcode` is set.
  - one-time-passcode is emailed to user's email.
  - User is presented with a new screen to enter one-time-passcode.
2. User then enters the one-time-passcode.
  - `one_time_passcode_expires_at` and `hashed_one_time_passcode` is set to empty.
  - `must_reset_password` is set to true.
  - User is logged in.
3. User is now forced to change their password.
4. User changes their password.
  - `must_reset_password` is set to false.

Reply:
Is there a situation where `hashed_one_time_passcode` could be set but `one_time_passcode_expires_at` could be empty?

Reply:
There shouldn't need to be.

Reply:
It would probably be a good idea then to enforce, at the schema level, that if `hashed_one_time_passcode` is `NOT NULL` then `one_time_passcode_expires_at` must also be `NOT NULL`.